streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/ColdPublisher.kt

```diff
@@ -27,5 +27,8 @@ class ColdPublisher<T>(private val executionBlock: ColdPublisherExecutionBlock<T
     override fun onNoSubscription() {
         super.onNoSubscription()
         cancellableManagerProvider.cancelPreviousAndCreate()
+        if (!completed) {
```

Comment:
Intriguing, why a cleanup only when not completed ? A completed publisher will not emit any more events right ?

Reply:
is it because the "onComplete" takes care of the cleanup in that case ?

Reply:
The reason behind this is to prevent resubscriber to receive a value "really" out of date.

If the publisher is completed, keep the values, reemit value and completion.
If the publisher is not completed, do not keep the values, resubscribe to the publisher the next time a subscriber come. The Shared (that is in fact a replayableShared) does the same thing.

Reply:
Got it, the onNoSubscription could happen before the onComplete State 🤦‍♂